### PR TITLE
React refactoring: Admin, AdminNicety, People

### PR DESCRIPTION
**Pull request by christalee**
These commits convert each component to use ES6 classes instead of React.createClass, and replace jQuery AJAX calls with fetch requests. The app behavior is unchanged.

### CODE REVIEW

File: src/components/AdminNicety.js [hunk 1]
```diff
@@ -23,19 +22,16 @@ class AdminNicety extends React.Component {
...
-            type: 'POST',
-            cache: false,
-            success: function () {
-                this.setState({noSave: true});
-            }.bind(this),
-            error: function (xhr, status, err) {
-                console.log(err)
-            },
-        });
+        fetch(this.props.admin_edit_api, {
```
Comment (jasonaowen) [inline src/components/AdminNicety.js:25]: Something to double-check, although I think this is already doing the right thing: does this `fetch` include all the necessary cookies? I think we're on the same origin, and the default behavior should be to send cookies in that case, but let's be certain.

File: src/components/Admin.js [hunk 2]
```diff
@@ -3,61 +3,65 @@ import $ from 'jquery';
...
+      error: function (xhr, status, err) {
+        console.error(this.props.people, status, err.toString());
+      }.bind(this),
+    });
+  }
+
+  componentDidMount = () => {
+    this.loadAllNiceties(function (data) {
+      this.setState({niceties: data});
+    }.bind(this));
```
Comment (jasonaowen) [inline src/components/Admin.js:26]: I think you will not need this `bind` any more, now that `loadAllNiceties` is an arrow function.

File: src/components/People.js [hunk 3]
```diff
@@ -9,180 +9,164 @@ import SaveButton from "./SaveButton";
...
+      setTimeout(function () {
+          this.setState({justSaved: false});
+      }.bind(this), 3000);
+  }
+
+  render() {
+      let leaving = this.generateRows(this.props.people.leaving);
+      let staying = this.generateRows(this.props.people.staying);
+      let faculty = this.generateRows(this.props.people.faculty);
+      const savePass = this.saveReady.bind(this);
```
Comment (jasonaowen) [inline src/components/People.js:101]: Likewise, `saveReady` is now an arrow function and does not need a `bind`, or to be in its own constant.

**Review by jasonaowen — APPROVED** [review]
This is looking good!

Thank you for pairing with me on testing the `fetch` changes, @christalee - that satisfies the comment I had below about cookies.

Please take a look at the calls to `bind` below, since this refactoring changes the behavior somewhat.

**Comment by christalee** [comment]
I've removed the unneeded `bind` calls, and converted a few more functions to arrows. The app builds and has all the usual functionality.

File: src/components/Admin.js [hunk 4]
```diff
@@ -9,17 +9,15 @@ class Admin extends React.Component {
   }
 
   loadAllNiceties = (callback) => {
-    $.ajax({
```
Comment (jasonaowen) [inline src/components/Admin.js]: Glad to be rid of this call to jQuery! I think this was the last reference to `$` in this file, so you can remove the import:

```
  2:8  warning  '$' is defined but never used  no-unused-vars
```

**Review by jasonaowen — APPROVED** [review]
Thanks for fixing, @christalee! I didn't mean for you to wait on me to merge - when I approve with suggestions (as I'm doing again, here), that is an invitation to merge after fixing whatever minor thing, or to ignore it if you disagree.

**Comment by jasonaowen** [comment]
Oh but you will have to rebase and resolve conflicts, I just noticed.

**Pull request merged**